src/main/java/fr/openmc/core/utils/text/TextManager.java

```diff
@@ -0,0 +1,132 @@
+        this.plugin = plugin;
+        this.defaultLanguage = defaultLanguage;
+        this.translationFolder = translationFolder;
+        this.loadedLanguages = new HashMap<>();
+    }
+
+    /**
+     * Returns a string corresponding to the specified path and the language.
+     *
+     * @param path  The path to the text, "default" for default language"
+     * @param language The language of the text
+     */
+    public String getText(String path, String language) {
+        FileConfiguration languageConfig = this.loadedLanguages.get(language);
+                this.loadedLanguages.get(this.defaultLanguage);
+        if (languageConfig != null || language == this.defaultLanguage) {
+            return languageConfig.getString(path, "Missing text for path: " + path);
+        } else {
+            return getText(path);
+        }
+    }
+
+    /**
+     * Returns a string corresponding to the specified path and the language and replaces the given placeholders with the values.
+     *
+     * @param path  The path to the text
+     * @param language The language of the text, "default" for default language"
+     * @param placeholders The placeholders you want to replace in pair with values ("player", player.getName())
+     */
+    public String getText(String path, String language, String... placeholders) {
+        return replacePlaceholders(getText(path, language), placeholders);
+    }
+
+    /**
+     * Returns a string corresponding to the specified path and the default language.
+     *
+     * @param path  The path to the text
+     */
+    public String getText(String path) {
+        return getText(path, this.defaultLanguage);
+    }
+
+
+    /**
+     * Loads the specified language if it is present in the translations' folder.
+     *
+     * @param language The language to load
+     */
+    public void loadLanguage(String language) {
+
+        File languageFile = new File(this.translationFolder, language + ".yml");
+
+        if (!languageFile.exists()) {
+            try {
+                this.plugin.saveResource(translationFolder.getPath() + language + ".yml", false);
+                plugin.getLogger().info("Language loaded : " + language);
+            }
+            catch (Exception ignored) {
+                plugin.getLogger().warning("Language " + language + " does not exist");
+            }
+        }
+
+        if (languageFile.exists()) {
+            this.loadedLanguages.put(language, YamlConfiguration.loadConfiguration(languageFile));
+            plugin.getLogger().info("Language " + language + " loaded");
+        }
+    }
+
+    /**
+     * Replaces the keys (between {}) in a String with the value Strings.
+     *
+     * @param text The string to modify
+     * * @param placeholders The placeholders you want to replace in pair with values ("player", player.getName())
+     */
+    public String replacePlaceholders(String text, String... placeholders) {
```

Comment:
Peut-être utiliser une map plutôt qu'un tableau de `String`.
```suggestion
    public String replacePlaceholders(String text, Map<String, String> placeholders) {
```
Comme ça, nous pourrions l'utiliser comme ceci
```java
instance.replacePlaceholders("mon super text {player}", Map.of(
        "player", player.getName()
))
```

Reply:
Je trouve l'original plus simple à utiliser

Reply:
Je ne peux pas résoudre cette conversation, je vous laisse faire.